velox/duckdb/conversion/tests/DuckParserTest.cpp

```diff
@@ -329,6 +329,106 @@ TEST(DuckParserTest, orderBy) {
+      return "UNBOUNDED PRECEDING";
+    case duckdb::BoundType::kUnboundedFollowing:
+      return "UNBOUNDED FOLLOWING";
+    case duckdb::BoundType::kPreceding:
+      return "PRECEDING";
+    case duckdb::BoundType::kFollowing:
+      return "FOLLOWING";
+    case duckdb::BoundType::kCurrentRow:
+      return "CURRENT ROW";
+  }
+  VELOX_UNREACHABLE();
+}
+
+const std::string parseWindow(const std::string& expr) {
+  auto windowExpr = parseWindowExpr(expr);
+  std::string concatPartitions = "";
+  int i = 0;
+  for (const auto& partition : windowExpr.partitionBy) {
+    concatPartitions += partition->toString();
+    if (i > 0) {
+      concatPartitions += " , ";
+    }
+    i++;
+  }
+  auto partitionString = windowExpr.partitionBy.empty()
+      ? ""
+      : fmt::format("PARTITION BY {}", concatPartitions);
+
+  std::string concatOrderBys = "";
+  i = 0;
+  for (const auto& orderBy : windowExpr.orderBy) {
+    concatOrderBys += fmt::format(
+        " {} {}", orderBy.first->toString(), orderBy.second.toString());
+    if (i > 0) {
+      concatOrderBys += " , ";
+    }
+    i++;
+  }
+  auto orderByString = windowExpr.orderBy.empty()
+      ? ""
+      : fmt::format("ORDER BY {}", concatOrderBys);
+
+  auto frameString = fmt::format(
+      "{} BETWEEN {}{} AND{} {}",
+      windowTypeString(windowExpr.frame.type),
+      (windowExpr.frame.startValue
+           ? windowExpr.frame.startValue->toString() + " "
+           : ""),
+      boundTypeString(windowExpr.frame.startType),
+      (windowExpr.frame.endValue ? " " + windowExpr.frame.endValue->toString()
+                                 : ""),
+      boundTypeString(windowExpr.frame.endType));
+
+  return fmt::format(
+      "{} OVER ({} {} {})",
+      windowExpr.functionCall->toString(),
+      partitionString,
+      orderByString,
+      frameString);
+}
+} // namespace
+
+TEST(DuckParserTest, window) {
+  EXPECT_EQ(
+      "row_number() AS c OVER (PARTITION BY \"a\" ORDER BY  \"b\" ASC NULLS LAST"
+      " RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)",
+      parseWindow("row_number() over (partition by a order by b) as c"));
+  EXPECT_EQ(
+      "row_number() AS a OVER (  RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)",
+      parseWindow("row_number() over () as a"));
+  EXPECT_EQ(
+      "row_number() AS a OVER ( ORDER BY  \"b\" ASC NULLS LAST "
+      "RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW)",
+      parseWindow("row_number() over (order by b) as a"));
+  EXPECT_EQ(
+      "row_number() OVER (PARTITION BY \"a\"  ROWS BETWEEN "
+      "UNBOUNDED PRECEDING AND CURRENT ROW)",
+      parseWindow(
+          "row_number() over (partition by a rows between unbounded preceding and current row)"));
+  EXPECT_EQ(
+      "row_number() OVER (PARTITION BY \"a\" ORDER BY  \"b\" ASC NULLS LAST "
+      "ROWS BETWEEN plus(\"a\",10) PRECEDING AND 10 FOLLOWING)",
+      parseWindow("row_number() over (partition by a order by b "
```

Comment:
nit: perhaps, modify one of the expressions to use DESC NULLS FIRST to make sure this clause is preserved